docs/bugpattern/LockOnNonEnclosingClassLiteral.md

```diff
@@ -1,18 +1,33 @@
-Lock on the class other than the enclosing class of the code block can
-unintentionally prevent the locked class or other classing using the same lock
-being used properly. A issue can be caused when the locked class being edited or
-deleted. And it can also be exhausting and time consuming for the others who is
-using the same class literal as lock to make sure the synchronized blocks can
-work properly as expected. Hence locking on the class other than the enclosing
-class of the synchronized code block is disencouraged by the error prone.
-Locking on the enclosing class or an instance is a preferred practice.
+Having a lock on a class, other than the enclosing class of the code block, can
+unintentionally prevent the locked class from being used properly when other
+classes effectively lock on the same resource. This can be time-consuming to
+handle from a maintainability perspective to ensure the `synchronized` blocks
+are working as expected. Hence, locking on a class other than the enclosing
+class of the `synchronized` code block is discouraged by Error Prone. Locking
+on the enclosing class or an instance is a preferred practice.
```

Comment:
Sorry, saw one more grammatical opportunity to be a bit more clear (preventing 'to handle [...] to ensure').

```suggestion
classes effectively lock on the same resource. From a maintainability
perspective, it can be time-consuming to ensure the `synchronized` blocks are
working as expected. Hence, locking on a class other than the enclosing class
of the `synchronized` code block is discouraged by Error Prone. Locking on the
enclosing class or an instance is a preferred practice.
```

Also, one of the reasons why this is discourages, can also be the opaqueness of such issues and it being hard to identify, which is not really captured well. At least, that is what 'time consuming and exhausting' made me think about. But I couldn't come up with the right words for that. Left it out for now, but if you feel different, feel free to tweak.

Reply:
Thanks for the suggestions! 😄